kmelia/kmelia-war/src/main/java/org/silverpeas/components/kmelia/control/KmeliaSessionController.java

```diff
@@ -758,15 +758,16 @@ public synchronized NodePK addSubTopic(NodeDetail nd, String alertType, String p
     nd.setCreatorId(getUserId());
     NodePK pk = getKmeliaService().addSubTopic(getNodePK(parentId), nd, alertType);
 
-    // by default, setting father's admin rights
-    // to preventing no more access
-    if (nd.haveRights()) {
+    // by default, sets father's admin rights in the case of specific rights
+    // to prevent no access to users having some rights with the father topic
+    NodeDetail created = getNodeHeader(pk.getId());
+    if (created.haveLocalRights()) {
       ProfileInst parentProfile;
       String profileAdmin = SilverpeasRole.ADMIN.getName();
       if (NodePK.ROOT_NODE_ID.equals(parentId)) {
         parentProfile = getProfile(profileAdmin);
       } else {
-        parentProfile = getTopicProfile(profileAdmin, parentId);
+        parentProfile = getRecursiveTopicProfile(profileAdmin, parentId);
```

Comment:
I don't understand why it is necessary to get the parent administrator profiles recursively.
If a user creates a folder, it means that he has an administrator profile on the folder he creates from. And therefore, the parent folder of the created folder necessarily has the administrator profile.
Am I missing something?

Reply:
A folder with specific rights can be created into a path of folders with, among them, a folder with specific rights. The goal here is to fetch all the administrative rights of this folder to set them as default administrator to the new folder. And we have to take into account this new folder is in a path of folders in which no one has specific rights: the administrators to copy for the new folder should be then the one of the root folder, id est the Kmelia instance.

Reply:
How do you get the situation where the new folder is created into a path where no one has specific rights?

Reply:
OK, Miguel gave me a phone call to explain to me the different cases.